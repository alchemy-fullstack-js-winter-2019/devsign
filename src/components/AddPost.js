import React from 'react';
import Header from './Header';
import Footer from './Footer';

export default function AddPost() {
  return (
    <>
    <Header/>
      <section>
        <h3>Add New Post</h3>
        <form>
          <label htmlFor="title">Post Title</label>
          <input type="text" name="title" id="title" size="30" placeholder="post title" required/>
          <label htmlFor="post">Enter new post here</label>
          <textarea  name="post" id="post" rows="10" cols="40">Another awesome content here...</textarea> 
          <button>Add</button>
        </form>
      </section>
      <Footer/>
    </>  
  );
}

