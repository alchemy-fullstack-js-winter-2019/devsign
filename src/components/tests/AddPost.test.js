import React from 'react';
import renderer from 'react-test-renderer';
import AddPost from '../AddPost';
import { MemoryRouter } from 'react-router';

describe('AddPost snapshot', () => {
  it('can match a snapshot', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <AddPost />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
