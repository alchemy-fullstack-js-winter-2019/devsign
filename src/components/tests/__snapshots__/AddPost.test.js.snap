// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPost snapshot can match a snapshot 1`] = `
Array [
  <p>
    My Blog
  </p>,
  <nav>
    <span>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </span>
    <span>
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </span>
    <span>
      <a
        href="/posts"
        onClick={[Function]}
      >
        Posts
      </a>
    </span>
    <span>
      <a
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </span>
  </nav>,
  <section>
    <h3>
      Add New Post
    </h3>
    <form>
      <label
        htmlFor="title"
      >
        Post Title
      </label>
      <input
        id="title"
        name="title"
        placeholder="post title"
        required={true}
        size="30"
        type="text"
      />
      <label
        htmlFor="post"
      >
        Enter new post here
      </label>
      <textarea
        cols="40"
        id="post"
        name="post"
        rows="10"
      >
        Another awesome content here...
      </textarea>
      <button>
        Add
      </button>
    </form>
  </section>,
  <p>
    Copyright 2019
  </p>,
]
`;
